test/integration/sds_dynamic_integration_test.cc

```diff
@@ -230,10 +233,14 @@ class SdsDynamicDownstreamIntegrationTest : public SdsDynamicIntegrationBaseTest
     });
 
     config_helper_.addConfigModifier([](envoy::config::bootstrap::v3::Bootstrap& bootstrap) {
-      // Add a static sds cluster
-      auto* sds_cluster = bootstrap.mutable_static_resources()->add_clusters();
-      sds_cluster->MergeFrom(bootstrap.static_resources().clusters()[0]);
+      // Add a static SDS cluster as the first cluster in the list.
+      // The SDS cluster needs to appear before the cluster that uses it for secrets, so that it
+      // gets initialized first.
+      bootstrap.mutable_static_resources()->add_clusters()->MergeFrom(
```

Comment:
so this is an interesting gotcha.  is this really just a test bug, or is this a real production issue if you don't know to magically order your clusters correctly?  I assume it's the latter, and if so  I think we'd be better off fixing with a fast-retry after start up in the production code rather than an ordering workaround in test.

cc @adisuissa for thoughts.

Reply:
I opened a bug that outlines the various issues: https://github.com/envoyproxy/envoy/issues/27702

Adi and I chatted about this on Friday, and felt that requiring cluster ordering for xDS clusters is a fair enough requirement for the time being (I will send a documentation PR update for this).

Adi is OOO this week, but we can chat in more detail when he gets back.